versioned/tiered/rocksdb/src/main/java/com/dremio/nessie/versioned/store/rocksdb/RocksL1.java

```diff
@@ -0,0 +1,214 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.versioned.store.rocksdb;
+
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.dremio.nessie.tiered.builder.L1;
+import com.dremio.nessie.versioned.Key;
+import com.dremio.nessie.versioned.impl.condition.ExpressionPath;
+import com.dremio.nessie.versioned.store.Entity;
+import com.dremio.nessie.versioned.store.Id;
+import com.dremio.nessie.versioned.store.StoreException;
+import com.google.protobuf.InvalidProtocolBufferException;
+
+/**
+ * A RocksDB specific implementation of {@link com.dremio.nessie.tiered.builder.L1} providing
+ * SerDe and Condition evaluation.
+ */
+class RocksL1 extends RocksBaseValue<L1> implements L1, Evaluator {
+
+  static final int SIZE = 43;
+  static final String COMMIT_METADATA = "metadataId";
+  static final String ANCESTORS = "ancestors";
+  static final String CHILDREN = "children";
+  static final String KEY_LIST = "keylist";
+  static final String INCREMENTAL_KEY_LIST = "incrementalKeyList";
+  static final String COMPLETE_KEY_LIST = "completeKeyList";
+  static final String CHECKPOINT_ID = "checkpointId";
+  static final String DISTANCE_FROM_CHECKPOINT = "distanceFromCheckpoint";
+
+  private Id metadataId; // commitMetadataId
+  private Stream<Id> parentList; // ancestors
+  private Stream<Id> tree; // children
+
+  private Stream<Key.Mutation> keyMutations; // keylist
+  private Id checkpointId; // checkpointId
+  private int distanceFromCheckpoint; // distanceFromCheckpoint
+  private Stream<Id> fragmentIds; // completeKeyList
+
+  RocksL1() {
+    super();
+  }
+
+  @Override
+  public L1 commitMetadataId(Id id) {
+    this.metadataId = id;
+    return this;
+  }
+
+  @Override
+  public L1 ancestors(Stream<Id> ids) {
+    this.parentList = ids;
+    return this;
+  }
+
+  @Override
+  public L1 children(Stream<Id> ids) {
+    this.tree = ids;
+    return this;
+  }
+
+  @Override
+  public L1 keyMutations(Stream<Key.Mutation> keyMutations) {
+    this.keyMutations = keyMutations;
+    return this;
+  }
+
+  @Override
+  public L1 incrementalKeyList(Id checkpointId, int distanceFromCheckpoint) {
+    this.checkpointId = checkpointId;
+    this.distanceFromCheckpoint = distanceFromCheckpoint;
+    return this;
+  }
+
+  @Override
+  public L1 completeKeyList(Stream<Id> fragmentIds) {
+    this.fragmentIds = fragmentIds;
+    return this;
+  }
+
+  @Override
+  public boolean evaluate(Condition condition) {
+    boolean result = true;
+    for (Function function: condition.getFunctions()) {
```

Comment:
It seems like this structure is common across all the evaluate() method impls, how can you refactor to reduce code duplication?

Reply:
Moved loop implementation up into RocksBaseValue and added abstract method to be implemented by child classes.